applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,709 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <stdio.h>
+#include <logging/log.h>
+#include <net/cloud.h>
+#include <nrf_modem_gnss.h>
+#if defined(CONFIG_SLM_AGPS)
+#include <net/nrf_cloud_agps.h>
+#endif
+#if defined(CONFIG_SLM_PGPS)
+#include <date_time.h>
+#include <net/nrf_cloud_pgps.h>
+#include <pm_config.h>
+#endif
+#if defined(CONFIG_SLM_CELL_POS)
+#include <net/nrf_cloud_cell_pos.h>
+#endif
+#include "slm_util.h"
+#include "slm_at_gnss.h"
+
+LOG_MODULE_REGISTER(gnss, CONFIG_SLM_LOG_LEVEL);
+
+#if defined(CONFIG_SLM_PGPS)
+#error "P-GPS not ready yet"
+#endif
+
+/**@brief GPS operations. */
+enum slm_gps_operation {
+	GPS_STOP,
+	GPS_START
+};
+
+#if defined(CONFIG_SLM_AGPS) || defined(CONFIG_SLM_PGPS) || defined(CONFIG_SLM_CELL_POS)
+
+#define NRF_CLOUD_ACCESS
+
+#define SERVICE_INFO_GPS \
+	"{\"state\":{\"reported\":{\"device\": {\"serviceInfo\":{\"ui\":[\"GPS\"]}}}}}"
+
+/**@brief nRF Cloud location service operations. */
+enum slm_location_operation {
```

Comment:
Is this used somewhere?

Reply:
A-GPS, P-GPS, CELL-POS AT command handler